### ISSUE TITLE
Plugin: require gives misleading error message

### ISSUE THREAD

**Issue by GitJaap** [octobercms/october#2298]
##### Expected behavior
If you enter a non-existing plugin name(or wrong casing in the 
```php
public $require = ['Rainlab.User']; //should be RainLab
```
and then run plugin:refresh, you should get an error saying that the required plugin(rainlab.user) is not found.

##### Actual behavior
What you get instead after running plugin:refresh is InvalidArgumentException Plugin (myname.myplugin) not found. Which gives no indication where you made the error. Also this error gets cached somehow, so if you correct the error and refresh again you still get plugin not found. Only after clearing the cache and fixing the error it is resolved. 

##### October build
351

I just wanted to share this with you since it costs me several hours to find out what was going wrong.

**Comment by gabsource** [comment]
`PluginManager::exists()` tests if the plugin is defined (ID found) and if it is not disabled. Plugins with missing dependencies are disabled automatically.

A possible way to display a more accurate message could be to add a test inside the _not exists_ `if` to throw an exception with a different message if the plugin is _disabled_.

Example :
```
        if (!PluginManager::instance()->exists($pluginName)) {
            if (PluginManager::instance()->isDisabled($pluginName)) {
                throw new \InvalidArgumentException(sprintf('Plugin "%s" has been manually disabled or some of its dependencies are missing as defined in the "$require" property.', $pluginName));
            }
            throw new \InvalidArgumentException(sprintf('Plugin "%s" not found.', $pluginName));
        }
```

Another way would be to allow resolving plugins ID in a case insensitive manner but that's probably not a good option as it would break the implicit _convention_ between PHP namespaces and plugins ID.

**Comment by daftspunk** [comment]
Thanks for reporting this, it is a duplicate of #1787. We will address it soon.
